Runtime/ui/painting/txt/mesh_generator.cs

```diff
@@ -196,24 +193,25 @@ class TextBlobMesh : PoolObject {
                     char a = text[this.textBlob.textOffset + i];
                     int code = a;
                     if (char.IsHighSurrogate(a)) {
-                        D.assert(i+1 < this.textBlob.textSize);
-                        D.assert(this.textBlob.textOffset+i+1 < this.textBlob.text.Length);
-                        char b = text[this.textBlob.textOffset+i+1];
+                        D.assert(i + 1 < this.textBlob.textSize);
+                        D.assert(this.textBlob.textOffset + i + 1 < this.textBlob.text.Length);
+                        char b = text[this.textBlob.textOffset + i + 1];
                         D.assert(char.IsLowSurrogate(b));
                         code = char.ConvertToUtf32(a, b);
-                    } else if (char.IsLowSurrogate(a) || EmojiUtils.isEmptyEmoji(a)) {
+                    }
+                    else if (char.IsLowSurrogate(a) || EmojiUtils.isEmptyEmoji(a)) {
                         continue;
                     }
+
                     var uvRect = EmojiUtils.getUVRect(code);
 
-                    var pos = this.textBlob.positions[i];
+                    var pos = this.textBlob.getPosition(i);
 
                     int baseIndex = vert.Count;
-                    vert.Add(new Vector3(pos.x + minX, pos.y + minY, 0));
-                    vert.Add(new Vector3(pos.x + maxX, pos.y + minY, 0));
-                    vert.Add(new Vector3(pos.x + maxX, pos.y + maxY, 0));
-                    vert.Add(new Vector3(pos.x + minX, pos.y + maxY, 0));
-
+                    vert.Add(new Vector3(pos + minX, minY, 0));
```

Comment:
The logic here is not clear.

why pos is a float rather than a real position like (x,y) ?  if this pos is really posX, maybe we should rename it to just posX to clearly match its meaning

Reply:
I just observed that pos.y is actually never used, and save some memory. I will change the name for clarity.